src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala

```diff
@@ -165,7 +165,24 @@ private[hyperspace] object IndexSummary {
       entry.derivedDataset.properties.columns.included,
       entry.numBuckets,
       entry.derivedDataset.properties.schemaString,
-      entry.content.root,
+      indexDirPath(entry),
       entry.state)
   }
+
+  /**
+   * This method extracts out the first index leaf directory path from a log entry. A leaf directory
+   * is defined as one with at least one leaf file in it.
+   *
+   * @param entry Index log entry.
+   * @return Path to the first leaf directory starting from the root.
+   */
+  private def indexDirPath(entry: IndexLogEntry): String = {
+    var root = entry.content.root
+    var indexDirPath = new Path(entry.content.root.name)
+    while (root.files.isEmpty) {
+      root = root.subDirs.head
```

Comment:
is DFS what you want? Or we need to do BFS?

Reply:
sorry I didn't understand. Could you explain further? Here I am doing dfs until the first directory with at least one leaf file is found. Not sure what the bfs logic would look like. (Code snippet would be helpful, thanks)

Reply:
if you want top-level, you want BFS if the index files are at different level. I guess you are assuming index files are at the same level, but let's not bring that assumption here. You can search online for BFS implementation.

Reply:
If you want to do this as a follow up, please add a "TODO" with the assumption, etc.

Reply:
Got it. But based on the current structure (tree structure with no duplciation), this is not a requirement anymore. Could you please cofnirm?
For now, I think this condition will suffice:
Find the bottom-most directory starting from root, where we find the first leaf files Or where there are multiple subdirectories.

This should give us our lowest common ancestor which is what I think you are referring.
Please let me know if the fix is good enough, otherwise I will add the TODO for investingating bfs.

Reply:
Seems fine to me for now.